csharp/src/SeedLang/Runtime/Value.cs

```diff
@@ -0,0 +1,220 @@
+// Copyright 2021 The Aha001 Team.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+
+namespace SeedLang.Runtime {
+  // The value type used in the SeedLang core components.
+  //
+  // Design consideration:
+  // 1) It's designed as a value type (struct) to avoid object creating frequently. It's wrapped by
+  //    a ValueWrapper and sent to the visualizer center when visualizers need be notified.
+  // 2) "in" keyword is used when passing Value as a parameter of functions to avoid copying.
+  // 3) "ref readonly" keywords are used when returning Value from a function to avoid copying.
+  internal readonly struct Value : IEquatable<Value> {
+    internal enum ValueType {
+      None,
+      Boolean,
+      Number,
+      String,
+      List,
+    }
+
+    private readonly ValueType _type;
+    private readonly double _number;
+    private readonly BoxValue _box;
```

Comment:
Per the actual usage of `_box` in the `Value` class, I guess a better name for `_box` could be `_valueOfReferenceType` or `_objOfReferenceType` or even `_nullableValue`. I personally like `_nullableValue`. See https://docs.microsoft.com/en-us/dotnet/api/system.nullable-1?view=net-6.0, A type is said to be nullable if it can be assigned a value or can be assigned null, which means the type has no value whatsoever. By default, all reference types, such as String, are nullable, but all value types, such as Int32, are not.

Reply:
Change to `HeapObject`